projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -230,6 +231,7 @@ class IIngesterPlugin:
     """
 
     IngestionMetadata = NewType("IngestionMetadata", Dict)
+    ExceptionsList = NewType("ExceptionsList", List)
```

Comment:
why list of exceptions? even with eventual exceptions chaining, would expect one instance when having one ingestion performed

Reply:
It is possible to have exceptions caused by pre-ingestion as well, so it seemed reasonable to have a list, as it is easier to work with.
However, it is not necessary to be a list. Can make it a single exception instance, if you think we don't need to catch multiple exceptions?

Reply:
**Case 1 - pre-processing exception**
Given pre-processors "pre1,pre2,pre3" configured, then
if "pre2" fails, "pp3" is not triggered, and no ingestion is expected to happen, 
therefore no post-processing will happen also. Agree?

**Case 2 - ingestion exception**
Given post-processors "post1,post2, post3" configured, then
if an exception occurs during ingestion,
"post1" receives 1 exception object only, then handles post-processing,
then "post2" receives 1 exception only (raised by ingestion),
then "post3" triggered receives the same ingestion exception only.
Agree?

**Case 3 - post-processing exception**
Given post-processors "post1,post2,post3" configured, then
common to pre-processors, any exception raised will interrupt the post-processing sequence.

**Case 3 scenario 1:**
If an user wants to implement a post-processing sequence, where 
each step has a prerequisite the success of the previous one,
raising exception on an exceptional event during "post1" guarantees post-handling atomicity. 
"post2" will execute only after "post1" successful execution.
Agree?

**Case 3 scenario 2:**
If an user wants to implement multiple independent post-processors,
this could be achieved by catching(+logging) exceptions, so the post-processing chain is not interrupted.
Agree?

**Case 3 scenario 3:**
If an user wants to implement stateful post-processing sequence, where 
"step2" will be aware of an eventual exception during "step1",
this could be achieved by passing any exception metadata via Optional[IngestionMetadata].
Agree?


If you agree on all, then no list of exceptions is needed - only one could happen during ingestion,
and all the post-processors are going to be aware of the ingestion state.

Reply:
Good point. Refactored, exception is now a single object, not a list. Thanks